pkg/service/auth.go

```diff
@@ -84,7 +89,15 @@ func NewAuthService(index index.Index, timeout time.Duration, maxHttpRequestBody
 // the response is compatible with the Dynamic Admission API
 func (a *AuthService) ServeHTTP(resp http.ResponseWriter, req *http.Request) {
 	ctx := context.New(context.WithTimeout(a.Timeout))
+	ctx = otel.GetTextMapPropagator().Extract(ctx, otel_propagation.HeaderCarrier(req.Header))
 	requestId := fmt.Sprintf("%x", sha256.Sum256([]byte(fmt.Sprint(req))))
+
+	attr := otel_trace.WithAttributes(otel_attr.String("authorino.request_id", requestId))
+	ctx, span := otel.Tracer("AuthService").Start(ctx, "ServeHTTP", attr)
```

Comment:
Excellent! So we start with 2 spans in the case of Authorino's [Raw HTTP Authorization interface](https://github.com/Kuadrant/authorino/blob/main/docs/architecture.md#raw-http-authorization-interface), plus the outer spans created by the instrumentation library.